src/components/ListItem.jsx

```diff
@@ -1,5 +1,42 @@
+import { useEffect, useState } from 'react';
 import './ListItem.css';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ name, itemId, listPath, purchaseDate }) {
+	/**TO DO: mark items as !isPurchased after 24 hours. Currently, I'm thinking
+	 * to subtract the purchaseDate from the current time at page load. Then check
+	 * if the difference is 24 hours. If less, item will still show as purchased. If
+	 * more than 24 hours difference, checkbox will be unchecked
+	 */
+	// let currentDate = new Date();
+	// console.log('check currentDate', currentDate);
+
+	// useEffect to check if item is purchased on page load so that the checkboxes are checked
+	useEffect(() => {
+		if (purchaseDate) {
+			setIsPurchased(true);
+		}
+	}, []);
+
+	// local state to keep track of purchased status
+	let [isPurchased, setIsPurchased] = useState(false);
```

Comment:
Interesting! I'm really not sure. I'm just used to putting my useEffect's at the top, so that's why it's in that order. But I'm curious if there is a better way to do it.

Reply:
Love this curiosity! 🔥 [Here’s](https://react.dev/reference/rules/rules-of-hooks) the documentation on the Rules of Hooks, which might help clarify things! — It mentions that you should always use Hooks at the top level of your React function to ensure that hooks are called in the correct order! 🚀